_posts/2020-10-27-solidity-0.8.x-preview.md

```diff
@@ -0,0 +1,192 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.x Preview Release'
+date: '2020-10-27'
+author: Solidity Team
+category: Releases
+---
+
+The Solidity team is currently preparing the changes for the next breaking release 0.8.x.
+
+Since this release contains a change that will affect many of you, we want to
+provide a preview release binary for everyone to try out so that you can give your feedback.
+
+This main change is the switch to checked arithmetic operations by default.
+This means that ``x + y`` will throw an exception on overflow. In other words: You do not need SafeMath anymore!
+
+The scope of 0.8.x is not fully specified, so if you want to influence what goes in and how,
+then either [create a new issue](https://github.com/ethereum/solidity/issues/new?template=feature_request.md)
+or voice your opinion on an [existing issue](https://github.com/ethereum/solidity/issues?q=is%3Aissue+is%3Aopen+label%3A%22breaking+change+%3Awarning%3A%22)
+
+The current tentatively accepted changes are those in the "Implementation Backlog" column of our project board
+that are labeled "Breaking Change": https://github.com/ethereum/solidity/projects/43?card_filter_query=label%3A%22breaking+change+%3Awarning%3A%22#column-9174474
+
+### Binaries!
+
+You can find the binaries at the following locations:
+
+TODO
+
+Note that all of these compilers will warn about them being a prerelease version, which is intentional.
+
+If you use solc-js, just copy the binary as ``soljson.js`` into the root directory of the solc-js npm module.
+
+For other ways to compile Solidity contracts, copy the binaries to the appropriate places.
+
+
+### Checked Arithmetic
+
+The "Checked Arithmetic" feature of Solidity 0.8.x consists of three sub-features:
+
+#### Revert on assertion failures and similar conditions instead of using the invalid opcode
+
+Prevously, internal errors like division by zero, failing assertions, array access out of bounds
+, etc., would result in an invalid opcode being executed. The idea was to distinguish
+these more critical errors from non-critical errors that lie outside of the domain
+of the Smart Contract programmer like invalid input, not enough balance for a token transfer,
+and so on. These non-critical errors use the ``revert`` opcode and optionally add an error description.
+
+The problem with the invalid opcode is that, in contrast to the ``revert`` opcode, it consumes all gas available.
+This makes it very expensive and you should try to avoid it at all cost.
```

Comment:
Why are we making this statement here?

Reply:
To explain why we are not using the invalid opcode anymore.

Reply:
> you should try to avoid it at all cost.

I think many would argue with this. There isn't a concrete concensus on this and now we're making a one sided statement?